nixos/modules/services/web-apps/ollama-webui.nix

```diff
@@ -0,0 +1,94 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let cfg = config.services.ollama-webui;
+
+in {
+  options = {
+    services.ollama-webui = {
+      enable = mkEnableOption (lib.mdDoc ''
+        Ollama-WebUI frontend service for the Ollama backend service.
+        It's a single page web application that integrates with Ollama to
+        run state-of-the-art AI large language models (LLM) locally with privacy
+        on your personal computer.
+        It's look and feel is similar to ChatGPT, supports different LLM models,
+        it stores chats (in the browser storage), supports image uploads,
+        Markdown and Latex rendering etc.See <https://github.com/ollama-webui/ollama-webui>.
+
+        The model names can be looked up on <https://ollama.ai/library> and are available in
+        varying sizes to fit your CPU, GPU, RAM and disk storage.
+
+        Required: This module requires the Ollama backend runner service that actually runs the
+        LLM models. The Ollama backend can be running locally or on a server; it's available
+        as a Nix package or a NixOS module.
+        The URL to the Ollama backend service can be set in this module, but overriden
+        later in the running Ollama-WebUI as well.
+
+        Optional: This module is configured to run locally, but can be served from a (home) server,
+        ideally behind a secured reverse-proxy.
+        Look at <https://nixos.wiki/wiki/Nginx> or <https://nixos.wiki/wiki/Caddy>
+        on how to set up a reverse proxy.
```

Comment:
do you think referencing the caddy file of the original repo would be a good idea ?
https://github.com/ollama-webui/ollama-webui/blob/main/Caddyfile.localhost